src/pivot-table/hooks/use-column-width.ts

```diff
@@ -142,89 +137,105 @@ export default function useColumnWidth(
     qEffectiveInterColumnSortOrder.length - qNoOfLeftDims > visibleTopDimensionInfo.length ? EXPAND_ICON_WIDTH : 0;
 
   /**
-   * Get the width of a leaf in the top grid. If there is no top grid, early return rightGridAvailableWidth
+   * Contains the unique column width values
+   * For a dimension, this is just one value, which every column will use
+   * For measures this means one value for each measure.
    */
-  const getLeafWidth = useCallback(
-    (info: ExtendedDimensionInfo | ExtendedMeasureInfo | undefined) => {
-      if (!info) return rightGridAvailableWidth;
+  const leafWidths = useMemo(() => {
+    const columnArray = topGridLeavesIsPseudo ? qMeasureInfo : [leafTopDimension];
+    const widths: number[] = [];
+    const autoColumnIndexes: number[] = [];
+    let sumAutoWidths = rightGridAvailableWidth;
 
-      const { qApprMaxGlyphCount, qFallbackTitle, columnWidth } = info;
-      const autoWidth = Math.max(rightGridAvailableWidth / layoutService.size.x, ColumnWidthValues.AutoMin);
-      let specifiedWidth = 0;
+    columnArray.forEach((col, idx) => {
+      if (col?.columnWidth) {
+        const {
+          columnWidth: { type, pixels, percentage },
+          qApprMaxGlyphCount,
+          qFallbackTitle,
+        } = col;
 
-      switch (columnWidth?.type) {
-        case ColumnWidthType.Pixels: {
-          specifiedWidth = getPixelValue(columnWidth.pixels);
-          break;
-        }
-        case ColumnWidthType.Percentage: {
-          specifiedWidth = getPercentageValue(columnWidth.percentage) * rightGridAvailableWidth;
-          break;
-        }
-        case ColumnWidthType.FitToContent: {
-          specifiedWidth = topGridLeavesIsPseudo
-            ? Math.max(estimateWidthForContent(qApprMaxGlyphCount), measureTextForColumnContent(qFallbackTitle))
-            : Math.max(
-                Math.max(...qMeasureInfo.map((m) => estimateWidthForContent(m.qApprMaxGlyphCount))),
-                estimateWidthForColumnContent(qApprMaxGlyphCount) + leavesIconWidth,
-              );
-          break;
-        }
-        case ColumnWidthType.Auto: {
-          // TODO: we might need to redo this in the pseudo dimension case, since if not all measure are set to auto
-          // you don't fill upp the width anyway
-          specifiedWidth = autoWidth;
-          break;
+        const addKnownWidth = (width: number) => {
+          widths[idx] = Math.min(ColumnWidthValues.PixelsMax, Math.max(ColumnWidthValues.PixelsMin, width));
+          // remove the width * number of instances of that width from the remaining width for auto columns
+          sumAutoWidths -= widths[idx] * (size.x / columnArray.length);
+        };
```

Comment:
you can put this out of iteration, the only value it needs is the idx which you can pass to it

Reply:
sure